### ISSUE TITLE
ddev delete images is keeping all versions of ddev-traefik-router

### ISSUE THREAD

**Issue by rpkoller** [ddev/ddev#6326]
### Preliminary checklist

- [X] I am using the latest stable version of DDEV (see [upgrade guide](https://ddev.readthedocs.io/en/stable/users/install/ddev-upgrade/))
- [X] I have searched [existing issues](https://github.com/ddev/ddev/issues)
- [X] I have checked the [troubleshooting guide](https://ddev.readthedocs.io/en/stable/users/usage/troubleshooting/)
- [X] I have run `ddev debug test` to include output below

### Output of `ddev debug test`

<details><summary>Expand `ddev debug test` diagnostic information</summary>

```
[COPY-PASTE HERE THE OUTPUT OF `ddev debug test`]
```
</details>

### Expected Behavior

I would expect after running `ddev delete images` only the image for the current version of ddev is left.  

### Actual Behavior

but instead the versions for the two previous versions are still there as well: 

![Screenshot 2024-06-19 at 03 15 31](https://github.com/ddev/ddev/assets/1665422/bd287fd0-2e2f-49ca-bda8-086d32437294)

### Steps To Reproduce

1. run `ddev delete images`
2. run `docker image ls`

### Anything else?

if you check the delete-images.go file the comment states that the check for router images "might" not work and has to be checked (See the screenshot) 

i've just checked all the preliminary checklist checkboxes. but i am not using the latest stable version of ddev i am using HEAD instead. but i'Ve checked the existing issues, and checking the troubleshooting guide doesnt apply here nor running the debug test makes much sense.

**Comment by agviu** [comment]
Yes I have experience this problem since traefik is the default router. I always run "ddev delete images" right after upgrading ddev, and the previous version of ddev-trafiek-router image is never deleted. I have to do it manually.
Checking the code in cmd/ddev/cmd/delete-images.go, where this is implemented, does not bring any light. The functiond deleteDdevImages searchs, through docker client, for images with "ddev/ddev-" tags and deletes them. Maybe the image is not found by docker client, or it is found but not deleted.

**Comment by rfay** [comment]
The reality is we should be deleting images based on a label, not based on the image name.
